common/scala/src/main/scala/whisk/http/BasicHttpService.scala

```diff
@@ -86,39 +94,58 @@ trait BasicHttpService extends HttpService with TransactionCounter {
      */
     def loglevelForRoute(route: String): Logging.LogLevel = Logging.InfoLevel
 
+    /** Rejection handler to terminate connection on a bad request. Delegates to Akka handler. */
+    val prioritizeRejections = recoverRejections { rejections =>
+        val priorityRejection = rejections.find(_.isInstanceOf[UnacceptedResponseContentTypeRejection]).getOrElse(null)
+
+        if (priorityRejection != null) {
+            Rejected(Seq(priorityRejection))
+        } else {
+            Rejected(rejections)
+        }
+    }
+
+    val toStrict = mapInnerRoute { innerRoute =>
+        extractRequest { req =>
+            onSuccess(req.toStrict(1.second)) { strictReq =>
+                mapRequest(_ => strictReq) {
+                    innerRoute
+                }
+            }
+        }
+    }
+
     /**
      * Receives a message and runs the router.
      */
-    def receive = runRoute(
+    def route: Route = {
         assignId { implicit transid =>
-            DebuggingDirectives.logRequest(logRequestInfo _) {
-                DebuggingDirectives.logRequestResponse(logResponseInfo _) {
-                    routes
+            handleRejections(customRejectionHandler) {
+                prioritizeRejections {
+                    DebuggingDirectives.logRequest(logRequestInfo _) {
+                        DebuggingDirectives.logRequestResult(logResponseInfo _) {
+                            toStrict {
+                                routes
+                            }
+                        }
+                    }
                 }
             }
-        })
+        }
+    }
+
+    def receive = {
+        case _ =>
+    }
```

Comment:
Yes, the Actor requires receive to be override. Tried not inheriting from the Actor, but it looks like we need access to the Actor context in controller.scala